### PR TITLE
Allow comma-separated multiple organizers

### PR DESCRIPTION
**Pull request by Camwyn**
🎫 https://central.tri.be/issues/65954

Allows users to filter the separator used for organizers

This allows for multiple organizers in these formats:
- single ID or name (with spaces)
- separator- or space-separated IDs
- separator-separated names
- separator-separated mix of names and IDs
- Has simple check for names that contain a separator - if an item doesn't return an ID, we concatenate it with the separator and the next item and test that (sorry, no names with two commas!)
- We no longer toss out results that have fewer items than the original split - that was silly and now we almost expect it (names with commas!)

**Caveats:**
Since names can contain spaces, we don't check for space-separated names (that could get a bit extreme). _Although the code will try if they use ' ' as their separator!_

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Usage of ELSE IF is discouraged; use ELSEIF instead | [File_Importer_Events.php:250](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/Tribe/Importer/File_Importer_Events.php#L250)
:x: | No space after opening parenthesis is prohibited | [File_Importer_Events.php:257](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/Tribe/Importer/File_Importer_Events.php#L257)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:257](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/Tribe/Importer/File_Importer_Events.php#L257)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:257](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/Tribe/Importer/File_Importer_Events.php#L257)
:x: | Comma not allowed after last value in single-line array declaration | [File_Importer_Events.php:266](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/Tribe/Importer/File_Importer_Events.php#L266)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [single-event.php:55](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/views/day/single-event.php#L55)
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$website' | [venue.php:46](https://github.com/moderntribe/the-events-calendar/blob/f91a445c5c52f2b60b3187fa615c0f93123f064d/src/views/modules/meta/venue.php#L46)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Usage of ELSE IF is discouraged; use ELSEIF instead | [File_Importer_Events.php:209](https://github.com/moderntribe/the-events-calendar/blob/d4293521e538253dfb71831acb35826c65b89922/src/Tribe/Importer/File_Importer_Events.php#L209)
:warning: | Usage of ELSE IF is discouraged; use ELSEIF instead | [File_Importer_Events.php:250](https://github.com/moderntribe/the-events-calendar/blob/d4293521e538253dfb71831acb35826c65b89922/src/Tribe/Importer/File_Importer_Events.php#L250)
:x: | Comma not allowed after last value in single-line array declaration | [File_Importer_Events.php:266](https://github.com/moderntribe/the-events-calendar/blob/d4293521e538253dfb71831acb35826c65b89922/src/Tribe/Importer/File_Importer_Events.php#L266)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
Check on the tests, we might need to update the tests, since before this was not the behavior we wanted.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | There must be a single space before the last parenthesis | [File_Importer_Events.php:249](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L249)
:x: | There must be a space between for and the first parenthesis '(' | [File_Importer_Events.php:266](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L266)
:x: | There must be a single space after the first square bracket | [File_Importer_Events.php:272](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L272)
:x: | There must be a single space before the last square bracket | [File_Importer_Events.php:272](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L272)
:x: | There must be a single space after the first square bracket | [File_Importer_Events.php:282](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L282)
:x: | There must be a single space before the last square bracket | [File_Importer_Events.php:282](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L282)
:x: | There must be a single space after the first square bracket | [File_Importer_Events.php:282](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L282)
:x: | Expected 1 space before &quot;+&quot;; 0 found | [File_Importer_Events.php:282](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L282)
:x: | Expected 1 space after &quot;+&quot;; 0 found | [File_Importer_Events.php:282](https://github.com/moderntribe/the-events-calendar/blob/f6a4d58178d9fb7e610013dd69e3551ce414da71/src/Tribe/Importer/File_Importer_Events.php#L282)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Importer/File_Importer_Events.php [hunk 1]
```diff
@@ -239,33 +239,105 @@ private function build_event_array( $event_id, array $record ) {
...
-			foreach ( $split as $possible_id_match ) {
-				$match[] = $this->find_matching_post_id( $possible_id_match, Tribe__Events__Organizer::POSTTYPE, 'any' );
+	/**
+	 * Filter allows user to customize the separator used for organizers
+	 * @since TBD
+	 *
+	 * @return mixed
+	 */
+	private function get_separator() {
+		return apply_filters( 'tribe_get_event_import_organizer_separator', ',' );
```
Comment (bordoni) [inline src/Tribe/Importer/File_Importer_Events.php]: Docblock

File: src/Tribe/Importer/File_Importer_Events.php [hunk 2]
```diff
@@ -239,33 +239,105 @@ private function build_event_array( $event_id, array $record ) {
...
+	 * @return array
+	 */
+	private function find_matching_organizer_id( $record ) {
+		$name      = $this->get_value_by_key( $record, 'event_organizer_name' );
+		$separator = $this->get_separator(); // We allow this to be filtered
+		$pattern   = '/' . $separator . '+/'; // build our regex pattern
+
+		// Test for $separator- or space-separated lists
+		if ( false !== stripos( $name, ',' ) || preg_match( '/\s+/', $name ) ) {
+			if ( is_numeric( preg_replace( '/\s+/', '', $name ) ) ) {
```
Comment (bordoni) [inline src/Tribe/Importer/File_Importer_Events.php]: I would love to see this matching go to a method too.

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Don't leave error_log lying around | [File_Importer_Events.php:324](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L324)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:324](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L324)
:x: | There must be a single space before the last parenthesis | [File_Importer_Events.php:324](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L324)
:x: | ! must be followed by a single space | [File_Importer_Events.php:329](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L329)
:x: | Don't leave error_log lying around | [File_Importer_Events.php:348](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L348)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:348](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L348)
:x: | There must be a single space before the last parenthesis | [File_Importer_Events.php:348](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L348)
:x: | No space after opening parenthesis is prohibited | [File_Importer_Events.php:357](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L357)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:357](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L357)
:x: | There must be a single space after the first parenthesis | [File_Importer_Events.php:357](https://github.com/moderntribe/the-events-calendar/blob/2a88df33a15ac8b1fde51ab7a6ca49e689fcbc07/src/Tribe/Importer/File_Importer_Events.php#L357)

> via jenkins/codesniffer-phpcs

**Comment by Camwyn** [comment]
Docblocks and tests added!

**Pull request merged**